File: packages/swift/Sources/CatalystSwift/Models/CatalystResult.swift
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

/// Generic `{"result": boolean}` response shape used by many mutation endpoints.
public struct CatalystResult: Decodable, Sendable, Equatable, Hashable {
  public let result: Bool
}

/// Generic `{"value": number}` response shape used by reaction toggle endpoints.
public struct CatalystReactionValue: Decodable, Sendable, Equatable, Hashable {
  public let value: Int
}

/// Generic `{"message": string}` response shape.
public struct CatalystMessage: Decodable, Sendable, Equatable, Hashable {
  public let message: String
}

File: packages/kotlin/src/main/kotlin/com/natsuneko/catalyst/models/CatalystResult.kt
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

package com.natsuneko.catalyst.models

import kotlinx.serialization.Serializable

/**
 * Generic `{"result": boolean}` response shape used by many mutation endpoints
 */
@Serializable
data class CatalystResult(val result: Boolean)

/**
 * Generic `{"value": number}` response shape used by reaction toggle endpoints
 */
@Serializable
data class CatalystReactionValue(val value: Int)

/**
 * Generic `{"message": string}` response shape
 */
@Serializable
data class CatalystMessage(val message: String)
